src/systemFlake.nix

```diff
@@ -19,12 +19,21 @@
     extraArgs = sharedExtraArgs;
   }
 
-, packagesBuilder ? null
-, defaultPackageBuilder ? null
-, appsBuilder ? null
-, defaultAppBuilder ? null
-, devShellBuilder ? null
-, checksBuilder ? null
+, outputsBuilder ? channels: {
+  packages = packagesBuilder channels;
+  defaultPackage = defaultPackageBuilder channels;
+  apps = appsBuilder channels;
+  defaultApp = defaultAppBuilder channels;
+  devShell = devShellBuilder channels;
+  checks = checksBuilder channels;
+}
```

Comment:
As you can imagine, this would have a couple consequences.
 - We create all the outputs whether users want them or not
 - I think `flake check` errors out from empty attrsets in defaultPackage and defaultApp

Reply:
Maybe we can use something like this instead?
```
        deprecatedBuilders = channels: {}
          // optionalAttrs (packagesBuilder != null) (packagesBuilder channels)
          // optionalAttrs (defaultPackageBuilder != null) (defaultPackageBuilder channels)
          // optionalAttrs (appsBuilder != null) (appsBuilder channels)
          // optionalAttrs (defaultAppBuilder != null) (defaultAppBuilder channels)
          // optionalAttrs (devShellBuilder != null) (devShellBuilder channels)
          // optionalAttrs (checksBuilder != null) (checksBuilder channels);
```